explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,405 @@
+    variants and variant-key attributes, the UA must execute the algorithm
+    written in
+    [HTTP Representation Variants spec](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+    to find the matching header.)
+    1. The UA checks whether all preload links which have matching
+    allowed-alt-sxg link header have matching (url and header-integrity)
+    parsedExchange which was copied from the referrer page.
+    1. If the check passes, in "linked resource fetch setup steps" for all
+    preload links which have matching allowed-alt-sxg link header the UA sets
+    request's stashed exchange to the matching parsedExchange so the resource
+    will be loaded from the cached signed exchange.
+
+# Detailed design discussion
+## Header integrity of signed exchange
+We propose a new `rel="allowed-alt-sxg"` link header with a `header-integrity`
+parameter. Publishers can declare that the subresource can be served by a signed
+exchange, using this link header in the inner response of the main resource
+signed exchange.
+
+```
+Link: <https://cdn.publisher.example/lib.js>;
+        rel="allowed-alt-sxg";
+        header-integrity="sha256-XXXXXX"
+```
+
+This `header-integrity` parameter is the SHA256 hash value of the
+*signedHeaders* value from the
+[application/signed-exchange format](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#application-signed-exchange)
+for integrity checking. This signedHeaders is *"the canonical serialization of
+the CBOR representation of the response headers of the exchange represented by
+the application/signed-exchange resource, excluding the Signature header
+field"*. So this value doesn’t change even if the publisher signs the content
+again or changes the signing key. This header-integrity value also guarantees
+the content body hasn't changed, because the signed headers are required to
+include a `Digest` header with a hash of the content body.
+
+The UA needs to check that this value of the prefetched subresource signed
+exchange is same as the header-integrity attribute of allowed-alt-sxg link
+header in the response from the publisher. This is intended to prevent the
+distributor from encoding a tracking ID into the subresource signed exchange.
+
+We can’t use the
+[SRI’s integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
+for this purpose, because SRI’s integrity only covers the content body and not
+any of the headers. So if the UA use the SRI’s integrity value in
+allowed-alt-sxg link header, we can use the subresource signed exchanges to
+track the users by changing content-type and detecting the image loading
+failure.
+
+## Multiple subresource signed exchanges
+If there are multiple subresource preloads that are provided by prefetches on
+the previous page in signed exchange format (example: script.js.sxg and
+image.jpg.sxg), the UA must check that there is no error in the any of the
```

Comment:
This "must" is covered above. Is there anything in this section that's not already covered? If there is, try to prune out the repetitive bits by referring to the algorithm. If not, just remove this section.